builtin/providers/cloudstack/provider.go

```diff
@@ -72,10 +91,42 @@ func Provider() terraform.ResourceProvider {
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	apiURL := d.Get("api_url").(string)
+	apiKey := d.Get("api_key").(string)
+	secretKey := d.Get("secret_key").(string)
+
+	if profile, ok := d.GetOk("profile"); ok {
```

Comment:
I would revert this to checking the `config` field as that makes more sense (for people reading the code)